programs/survival/showbiome.sc

```diff
@@ -0,0 +1,71 @@
+// shows biome name on the blocks around the player. useful for marking out rivers, beaches, soul sand valley and stuff.
+
+// if the script crashes your multiplayer world, do one of the two
+// 1. update carpet to latest version
+// 2. turn this to true 
+global_carpet_crash_fix = false;
+
+// change this global_radius to improve fps if you on a potato. 4 works decently
+global_radius = 4;
+
+__config() -> {
+    'stay_loaded'->true,
+    'commands' -> {
+        '' -> _() -> print(format('r Missing arguments. Correct syntax /showbiome <toggle>')),
+        '<toggle>' -> '__toggle_markings',
+        'set radius <radius>' -> _(radius) -> global_radius = radius
+    },
+    'arguments' -> {
+        'toggle' -> {'type' -> 'text', 'options' -> ['on', 'off']},
+        'radius' -> {'type' -> 'int', 'min' -> 1, 'max' -> 8, 'suggest' -> [2, 4, 8]}
+    }
+};
+
+global_player = {};
+global_dimension_map = {'overworld' -> 'foliage_color', 'the_nether' -> 'fog_color', 'the_end' -> 'sky_color'};
+
+__make_markers() -> (
```

Comment:
I missed this last time, but this doesn't seem to be multiplayer safe? not sure, since I didn't test, but `player()` will grab any player in the server when loading this app in `global_scope`, so what you'd need to do is iterate over the player list and run this funciton for each player (with toggle on).

Reply:
I have tested it in multiplayer and so far it only grabs the player who invokes the script. Because according to documentation player() does that